daffodil-lib/src/main/scala/org/apache/daffodil/lib/cookers/EntityReplacer.scala

```diff
@@ -774,18 +763,14 @@ class NonEmptyListOfStringLiteral(pn: String, allowByteEntities: Boolean)
 
   override def testCooked(cookedList: List[String], context: ThrowsSDE) = {
     context.schemaDefinitionUnless(
-      cookedList.exists { _.length > 0 },
+      cookedList.length > 0,
       "Property dfdl:%s cannot be empty string. Use dfdl:nilValue='%%ES;' for empty string as nil value.",
```

Comment:
This messsage mentions dfdl:nilValue but at least in theory it is not the only property that would use this cooker.

Just drop the property name. I think the message is ok without it. Or pass it in as a parameters somehow.

Reply:
Changed to use `%s` and `propName` so the error message should stay the same so not break existing tests.

Semi-related, in making these changes, I noticed a lot of duplication in our cookers (e.g. these multiple nil cookers). I'm thinking at some point we may want to refactor these. For example, we have some really complicated cookers that are like:

```scala
object ChoiceDispatchKeyCooker extends StringLiteralNonEmptyNoCharClassEntitiesNoByteEntities()
```
I imagine we can replace these just just a bunch of mixins, e.g.

```scala
object ChoiceDispatchKeyCooker extends StringLiteral
 with NonEmptyLiteral
 with NoCharClassEntities
 with NoByteEntities
```

So we just create a bunch of tiny specific traits that are easy to write and verify, and stack them all together. It makes the cooker definitions a bit longer, but makes it much easier to fix/change them without having to create new 50+ character classes when a new property needs something slightly different.

For another day though.